### PR TITLE
L2 Staking and Voting

### PR DESCRIPTION
**Pull request by soloseng**
### Description

Re-enabled staking and voting for validators. This PR does **NOT** address validator election and staking rewards distribution.

### Other changes

- Updated the UsingPrecompile.sol contract to calculate the epochSize.
- Moved Election test contract to `../governance/voting` so that it is included in CI voting tests

### Tested

Updated unit tests to simulate L2

### Related issues

- Fixes https://github.com/celo-org/celo-blockchain-planning/issues/327

### CODE REVIEW

File: packages/protocol/contracts/governance/Validators.sol [hunk 1]
```diff
@@ -597,7 +594,6 @@ contract Validators is
    *         the last time the group was slashed.
    */
   function resetSlashingMultiplier() external nonReentrant {
```
Comment (soloseng) [inline packages/protocol/contracts/governance/Validators.sol:596]: Made available on L2 to ensure that the value could be set to 1. 

Due to the contract size, it is difficult to add additional logic that would always return 1 on L2, without significantly rewriting the contract.

File: packages/protocol/contracts/governance/Validators.sol [hunk 2]
```diff
@@ -755,7 +751,6 @@ contract Validators is
    * @param account The group to fetch slashing multiplier for.
    */
   function getValidatorGroupSlashingMultiplier(address account) external view returns (uint256) {
```
Comment (soloseng) [inline packages/protocol/contracts/governance/Validators.sol:753]: DITTO: Can't modify contract to return 1 on L2 without going over `maxCodeSize`. Enabled `resetSlashingMultiplier` on L2 instead.

File: packages/protocol/contracts/governance/Validators.sol (hunk 1 above)
Comment (pahor167) [reply packages/protocol/contracts/governance/Validators.sol:596]: Should we create task for this ?

**Review by pahor167 — APPROVED** [review]

File: packages/protocol/contracts/governance/Validators.sol (hunk 1 above)
Comment (soloseng) [reply packages/protocol/contracts/governance/Validators.sol:596]: This is a function that is usually called by the validators. 

So if any validator is slashed at the L2 transition time, they will need to reset their multiplier like they would normally have to do on L1.

**Pull request merged**